woocommerce/woocommerce-android
# [Coupons] Restrictions: Allowed Emails Button UI and Navigation to Allowed Emails Screen

Part of: #5540


### Description
This PR adds a `WCFullWidthTextButton` component, that is then used for the "Allowed emails" button on the Coupon Restrictions screen, like on the screenshot below. The component optionally allows inline text and chevron.

<img width="404" alt="Screen Shot 2022-06-05 at 17 21 15" src="https://user-images.githubusercontent.com/266376/172045988-551cedd4-ae85-4080-8f9e-18ead4664496.png">

This PR also adds an empty Email Restriction screen, with its related Fragments, View Model, and Compose screen component.

Finally, this PR also adds navigation from Coupon Restrictions screen to Email Restriction screen, and data passing and handling between the two screens. Since the Email Restriction screen is still empty, the data passing / handling does not do anything and do not need to be tested yet. They will be tested in a future PR.

### Testing instructions


1. In wp-admin **Marketing > Coupons**, create two Coupons: One without email restriction and one with. To add the restriction, create a Coupon and go to **Usage Restrictions > Allowed emails**.
2. Go to the app, make sure Coupons beta toggle is enabled.
3. Open the details for the Coupon without email restriction, 
4. Click on the Menu, then click on "Edit coupon", then click "Usage Restrictions" to open Usage Restrictions screen.
5. Find the "Allowed emails" button and make sure the inline text says "No Restrictions".
6. Go back to Coupons list screen,
7. Open the details for the Coupon with email restriction,
8. Click on the Menu, then click on "Edit coupon", then click "Usage Restrictions" to open Usage Restrictions screen.
9. Find the "Allowed emails" button and make sure the inline text shows the email value (likely shortened due to limited space).
10. Tap the "Allowed emails" button and make sure it opens an empty Screen with "Allowed emails" title.
11. Tap the "X" button and make sure it returns to the Coupon Restrictions screen.

### Images/gif

 
| Button when there's no email restriction | Button when there's some email restriction |
|-|-|
| <img width="378" alt="Screen Shot 2022-06-06 at 10 36 24" src="https://user-images.githubusercontent.com/266376/172090440-b3c9158a-c8ec-4237-9816-20ce56191722.png">  | <img width="378" alt="Screen Shot 2022-06-06 at 10 36 05" src="https://user-images.githubusercontent.com/266376/172090443-968d6d10-9249-40f9-bc92-887882e6372d.png">  |
